src/dev/build/tasks/clean_tasks.js

```diff
@@ -84,3 +84,28 @@ export const CleanExtraBinScriptsTask = {
     }
   }
 };
+
+export const CleanExtraBrowsersTask = {
+  description: 'Cleaning extra browsers from platform-specific builds',
+
+  async run(config, log, build) {
```

Comment:
Haven't tested - so don't take the code literally, but wondering if something like this would be easier to read/follow:

```
async run(config, log, build) {
    const reportingPath = 'node_modules/x-pack/plugins/reporting';
    const browsers = '{.chromium,.phantom}';
    const builds = ['darwin', 'macosx', 'linux', 'win32', 'windows'];

    function browserPathsWithout(platform, exclude = []) {
      return build.resolvePathForPlatform(
        platform,
        reportingPath,
        browsers,
        _.without(builds, exclude)
      );
    }
    for (const platform of config.getPlatforms()) {

      if (platform.isWindows()) {
        await deleteAll(log, [browserPathsWithout(platform, ['win32', 'windows'])]);
      } else if (platform.isMac()) {
        await deleteAll(log, [browserPathsWithout(platform, ['darwin', 'macosx'])]);
      } else if (platform.isLinux()) {
        await deleteAll(log, [browserPathsWithout(platform, ['linux'])]);
      }
    }
```